subsys/net/lib/dns/mdns_responder.c

```diff
@@ -217,23 +247,24 @@ static int send_response(struct net_context *ctx,
 	socklen_t dst_len;
 	int ret;
 
+	ret = setup_dst_addr(ctx, pkt, &dst, &dst_len);
+	if (ret < 0) {
+		NET_DBG("unable to set up the response address");
+		return ret;
+	}
+
 	if (qtype == DNS_RR_TYPE_A) {
 #if defined(CONFIG_NET_IPV4)
```

Comment:
It looks like we could do `if (IS_ENABLED(CONFIG_NET_IPV4) && qtype == DNS_RR_TYPE_A) {` here too, and same thing for IPv6 branch.

Reply:
Cool, done as well, left the error as `-EINVAL` since the check here is about DNS RR.